ci/run-ci.py

```diff
@@ -0,0 +1,474 @@
+    cmd = f'gcloud container clusters create {CLUSTER_NAME} '\
+          f'--cluster-version {CLUSTER_VERSION} '\
+          f'--machine-type {CLUSTER_MACHINE_TYPE} '\
+          f'--service-account {SERVICE_ACCOUNT} '\
+          f'--num-nodes=1 '\
+          f'--zone={CLUSTER_ZONE} '\
+          f'--project={CLUSTER_PROJECT} '\
+          f'--async'
+    call(cmd)
+
+
+def delete_cluster_async():
+    wait_for_cluster()
+    print('# Delete cluster')
+    cmd = f'yes | gcloud container clusters delete {CLUSTER_NAME} --zone ' \
+          f'{CLUSTER_ZONE} --project {CLUSTER_PROJECT} --async --quiet'
+    call(cmd)
+
+
+def wait_for_cluster():
+    print('# Waiting for GKE cluster to be ready ...', end='')
+
+    while True:
+        output = call_output(
+            f'gcloud container clusters list --filter="name={CLUSTER_NAME}" --project {CLUSTER_PROJECT}',
+            print_cmd=False
+        )
+
+        try:
+            status = output.split('\n')[1].split(' ')[-1]
+            if status not in ['RUNNING', 'PROVISIONING']:
+                raise Exception(f'Unknown status {status}')
+        except Exception as e:
+            print('\nFATAL: Error retrieving cluster status. Output was:')
+            print(output)
+            raise(e)
+
+        if status == 'RUNNING':
+            print('done.')
+            break
+
+        print('.', end='', flush=True)
+        time.sleep(5)
+
+
+def setup_tiller():
+    print('\n# Set up Tiller')
+    call(f'kubectl --context {KUBE_CONTEXT} create serviceaccount --namespace kube-system tiller')
+    call(f'kubectl --context {KUBE_CONTEXT} create clusterrolebinding tiller-cluster-rule ' +
+         f'--clusterrole=cluster-admin --serviceaccount=kube-system:tiller')
+    call(f'helm --kube-context {KUBE_CONTEXT} init --service-account tiller --upgrade --wait')
+
+
+def clone_repos():
+    if os.path.exists(SOURCE_DIR):
+        shutil.rmtree(SOURCE_DIR)
+
+    os.makedirs(SOURCE_DIR)
+
+    print('\n# Clone repos')
+    commit_backend = clone_substra_backend()
+    commit_hlf = clone_hlf_k8s()
+    commit_tests = clone_substra_tests()
+
+    print(
+        f'\nCommit hashes:\n'
+        f'- substra-backend: \t{commit_backend}\n'
+        f'- hlf-k8s: \t\t{commit_hlf}\n'
+        f'- substra-tests: \t{commit_tests}\n'
+    )
+    return [
+        {'name': 'hlf-k8s',
+         'images': ['hlf-k8s'],
+         'commit': commit_hlf,
+         'branch': HLF_K8S_BRANCH},
+        {'name': 'substra-backend',
+         'images': ['substra-backend', 'celeryworker', 'celerybeat', 'flower'],
+         'commit': commit_backend,
+         'branch': SUBSTRA_BACKEND_BRANCH},
+        {'name': 'substra-tests',
+         'images': ['substra-tests'],
+         'commit': commit_tests,
+         'branch': SUBSTRA_TESTS_BRANCH},
+    ]
+
+
+def clone_repository(dirname, url, branch, commit=None):
+    call(f'git clone -q --depth 1 {url} --branch "{branch}" {dirname}')
+
+    if commit is None:
```

Comment:
it's a bit confusing for me here if the commit is given do we checkout the commit? or just return it? what is the purpose of passing a commit to this function?

Reply:
I think this was done to eventually allow specifying a commit hash at the top level of the script, but it wasn't finished. @Kelvin-M would know more

Reply:
Yes